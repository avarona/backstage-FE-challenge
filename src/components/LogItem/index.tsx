import React from 'react';

export type Props = {
  // could be global declaration if logging other types of items
  item: {
    datetime: string;
    value: string;
    number: string;
    occurrences: string;
    last_datetime: string;
  }
}

type Keys = Array<keyof Props['item']>;

const LogItem = ({ item }: Props) => {
  const itemProperties = Object.keys(item).sort() as Keys;

  return (
    <ul>
      {itemProperties.map((key, i) => {
        return <li key={i}>{key}: {item[key]}</li>
      })}
    </ul>
  )
};

export default LogItem;
