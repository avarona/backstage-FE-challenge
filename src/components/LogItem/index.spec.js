import React from 'react';
import { shallow } from 'enzyme';
import LogItem from './';

describe('LogItem', () => {
  const item = {
    datetime: 'what',
    value: 'am',
    number: 'I',
    occurrences: 'doing',
    last_datetime: '?',
  }

  const wrapper = shallow(<LogItem item={item} />);

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('is a list of properties', () => {
    expect(wrapper.find('ul').exists()).toBe(true);
    expect(wrapper.find('li').exists()).toBe(true);
  });
});
