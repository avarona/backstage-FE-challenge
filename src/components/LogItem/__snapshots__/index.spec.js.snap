// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItem renders 1`] = `"<ul class=\\"logItemContainer\\"><li><strong>datetime:</strong> what</li><li><strong>last_datetime:</strong> ?</li><li><strong>number:</strong> I</li><li><strong>occurrences:</strong> doing</li><li><strong>value:</strong> am</li></ul>"`;
